test/insert.js

```diff
@@ -0,0 +1,49 @@
+var Pipeline = require( '..' )
+var assert = require( 'assert' )
+var fs = require( 'fs' )
+var path = require( 'path' )
+var Stream = require( 'stream' )
+
+describe( 'Pipeline.insert()', function() {
+
+  it( 'can insert a stream at specific indices', function( done ) {
```

Comment:
IMHO there should be at least two transforms, each doing something different, so that you can check the order they're being applied in is correct. Because with only a lowercase transform, these pipelines are all equivalent:
```
lowercase
do-nothing | lowercase
lowercase | do-nothing
do-nothing | lowercase | do-nothing
do-nothing | lowercase | do-nothing | lowercase | lowercase
```
I guess you could modify the test-text to start out in MixedCase, and then you could have both `lowercase` and `uppercase` transforms?